diff -Naur dmenu-5.0/Makefile dmenu-5.0.tpg/Makefile
--- dmenu-5.0/Makefile	2020-09-02 16:37:07.000000000 +0000
+++ dmenu-5.0.tpg/Makefile	2021-03-22 11:00:26.471943280 +0000
@@ -23,10 +23,10 @@
 $(OBJ): arg.h config.h config.mk drw.h
 
 dmenu: dmenu.o drw.o util.o
-	$(CC) -o $@ dmenu.o drw.o util.o $(LDFLAGS)
+	$(CC) $(CFLAGS) -o $@ dmenu.o drw.o util.o $(LDFLAGS)
 
 stest: stest.o
-	$(CC) -o $@ stest.o $(LDFLAGS)
+	$(CC) $(CFLAGS) -o $@ stest.o $(LDFLAGS)
 
 clean:
 	rm -f dmenu stest $(OBJ) dmenu-$(VERSION).tar.gz
diff -Naur dmenu-5.0/config.mk dmenu-5.0.tpg/config.mk
--- dmenu-5.0/config.mk	2020-09-02 16:37:07.000000000 +0000
+++ dmenu-5.0.tpg/config.mk	2021-03-22 10:59:24.331232470 +0000
@@ -23,9 +23,9 @@
 LIBS = -L$(X11LIB) -lX11 $(XINERAMALIBS) $(FREETYPELIBS)
 
 # flags
-CPPFLAGS = -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700 -D_POSIX_C_SOURCE=200809L -DVERSION=\"$(VERSION)\" $(XINERAMAFLAGS)
-CFLAGS   = -std=c99 -pedantic -Wall -Os $(INCS) $(CPPFLAGS)
-LDFLAGS  = $(LIBS)
+CPPFLAGS += -D_GNU_SOURCE -DVERSION=\"$(VERSION)\" $(XINERAMAFLAGS)
+CFLAGS   := $(CFLAGS) -Wall $(INCS) $(CPPFLAGS)
+LDFLAGS  += $(LIBS)
 
 # compiler and linker
 CC = cc
